### PR TITLE
[WIP] Individual config in KafkaChannel

### LINKED ISSUE #1005: KafkaChannel Configuration option on a per object base
**Problem**

We currently have a global configuration for _one_ dedicated Apache Kafka cluster:
https://github.com/knative-sandbox/eventing-kafka/blob/main/config/channel/consolidated/configmaps/kafka-config.yaml#L24

this forces _all_ `KafkaChannel` objects to be connected against a single kafka cluster.

This does not fly well, when offering "Knative Kafka" as a service, where customers have desire for individual Kafka cluster usage, on a per channel bases.

The global default is a convenient setting for on prem, but not for more cloud-centric cases 

## Proposal

Adding the following fields to the `KafkaChannel` spec

- `bootstrapServers`
-  `authSecretName` (assumes that the actual secret is living in the namespace of the `KafkaChannel` object)

**[Persona:](https://github.com/knative/eventing/blob/master/docs/personas.md)**
Which persona is this feature for?

**Exit Criteria**
A measurable (binary) test that would indicate that the problem has been resolved.

**Time Estimate (optional):**
How many developer-days do you think this may take to resolve?

**Additional context (optional)**
Add any other context about the feature request here.

### PR DESCRIPTION
**Pull request by aliok**
Fixes #1005

## Proposed Changes

-
-
-

**Release Note**

```release-note

```

**Docs**

### CODE REVIEW

File: pkg/apis/messaging/v1beta1/kafka_channel_types.go [hunk 1]
```diff
@@ -64,6 +64,22 @@ var (
 
 // KafkaChannelSpec defines the specification for a KafkaChannel.
 type KafkaChannelSpec struct {
+	// BootstrapServers is the Kafka Channel bootstrap servers. When not provided, the implementation will
+	// fall back to the configmap in the config field. If there are also no bootstrapServers defined in that,
```
Comment (matzew) [inline pkg/apis/messaging/v1beta1/kafka_channel_types.go:68]: for the future: do we envision to retire the global config, all together?

**Review by pierDipi — COMMENTED** [review]
LGTM

can you run `hack/update-codegen.sh`?

File: pkg/apis/messaging/v1beta1/kafka_channel_types.go [hunk 2]
```diff
@@ -64,6 +64,22 @@ var (
...
+
+	// Auth configurations.
+	// +optional
+	Auth *Auth `json:"auth,omitempty"`
+
+	// Config is a KReference to the configuration that specifies
+	// configuration options for this Channel. For example, this could be
+	// a pointer to a ConfigMap.
+	// +optional
+	Config *duckv1.KReference `json:"config,omitempty"`
```
Comment (pierDipi) [inline pkg/apis/messaging/v1beta1/kafka_channel_types.go:81]: what's the use case for this?

**Comment by matzew** [comment]
@aliok with the on-going data-plane isolation, I think we should keep this proposal also in our backheads, as this allows individual configuration of the kafka channel's kafka (similar to the `broker.spec.config`).

Currently there is no MT usage o the channel, due to this global setting, and no individual config for kafka cluster

**Comment by matzew** [comment]
/close
the channel has been moved over to the `eventing-kafka-broker` repo

**Pull request closed without merging**